signing-scheme.md

```diff
@@ -0,0 +1,97 @@
+# Signing Scheme
+
+Signatures are primarily used to provide a consumer of an artifact the following guarantees - integrity (the signed artifact was not tampered after signing it), and authenticity (the artifact was indeed signed by the entity who claims to have signed it).
+X.509 PKI based identities are commonly used to for signing artifacts.
+It has well established standards for issuing and representing identities (CAs and certificates) which sign artifacts, mechanism to establish authenticity using trust stores, and support for managing key lifetimes and rotation, and revocation of identities.
+There is interest in the community to support other ways to establish integrity and authenticity, based on other systems and techniques (Notary TUF, ledger based).
+These approaches also can provide different feature sets than ones traditionally provided by standard X.509 PKI based signing.
+E.g. TUF is geared for software update systems and provides a freshness guarantee.
+
+Notary v2 will initially support X.509 PKI identity based signing, but provide the flexibility for additional systems through an abstraction called Signing Scheme.
+The Signing Scheme covers aspects of signature generation and verification, and formalizes the feature set (guarantees) provided by the signature produced using a signing scheme.
```

Comment:
Do we need to define trust policy type based on signing scheme?

Reply:
Current trust policy is optimized for known signing schemes, we don't know what policy changes will be required to support future signing schemes, which is addressed by policy versioning.